R/sync.R

```diff
@@ -136,7 +136,9 @@ latticeView <- function(...,
                 Array.prototype.map.call(
                  document.querySelectorAll(".leaflet"),
                    function(ldiv){
-                     leaf_widgets[ldiv.id] = HTMLWidgets.find("#" + ldiv.id).getMap();
+                     if (ldiv.id.indexOf("htmlwidget-") !== -1) {
+                        leaf_widgets[ldiv.id] = HTMLWidgets.find("#" + ldiv.id).getMap();
```

Comment:
This solves the problem in most cases, but there is a potential case where a user might manually specify an `id` different from this. I think a more robust solution will be something like this.

```
if( HTMLWidgets.find("#" + ldiv.id) && HTMLWidgets.find("#" + ldiv.id).getMap() ) {
  leaf_widgets[ldiv.id] = HTMLWidgets.find("#" + ldiv.id).getMap();
}
```

I am happy so submit a change, but I generally like for the original submitter to get attribution for their diligence and contribution, so I would prefer that you make change (if you agree) and modify your pull.

Also, I think because of the not-sure-it-is-a-bug behavior in Shiny with `renderUI`, I think the potentially better solution will be to use `renderUI` for all your leaflets if using with `sync`.  Nevertheless we should definitely handle this case, and I appreciate your discovery and work.